src/effects/effects_base/internal/effectsprovider.cpp

```diff
@@ -182,3 +187,241 @@ muse::Ret EffectsProvider::performEffect(AudacityProject& project, Effect* effec
 
     return success ? muse::make_ok() : muse::make_ret(Ret::Code::UnknownError);
 }
+
+static TransportSequences MakeTransportTracks(
+    TrackList& trackList, bool selectedOnly, bool nonWaveToo = false)
+{
+    TransportSequences result;
+    {
+        const auto range = trackList.Any<WaveTrack>()
+                           + (selectedOnly ? &Track::IsSelected : &Track::Any);
+        for (auto pTrack : range) {
+            result.playbackSequences.push_back(
+                StretchingSequence::Create(*pTrack, pTrack->GetClipInterfaces()));
+        }
+    }
+    if (nonWaveToo) {
+        const auto range = trackList.Any<const PlayableTrack>()
+                           + (selectedOnly ? &Track::IsSelected : &Track::Any);
+        for (auto pTrack : range) {
+            if (!track_cast<const SampleTrack*>(pTrack)) {
+                if (auto pSequence
+                        =std::dynamic_pointer_cast<const OtherPlayableSequence>(
+                              pTrack->shared_from_this())
+                        ) {
+                    result.otherPlayableSequences.push_back(pSequence);
+                }
+            }
+        }
+    }
+    return result;
+}
+
+static void EffectPreview(EffectBase& effect,
+                          EffectSettingsAccess& access, std::function<void()> updateUI, bool dryOnly)
+{
+    auto cleanup0 = effect.BeginPreview(access.Get());
+
+    // These are temporary state in the Effect object that are meant be moved to
+    // a new class EffectContext
+    const auto numTracks = effect.mNumTracks;
```

Comment:
Many of these variables, such as mProgress, are defined but not used in the scope of the EffectPreview function or used once. Consider removing or using these variables to avoid unused-variable warnings.

Reply:
https://github.com/audacity/audacity/pull/7537#discussion_r1808779273

Reply:
I prefer like Igor that these methods are guaranteed "left untouched" (with formatting tolerated), which means that their review can be skipped and time saved.